app/models/task_status.rb

```diff
@@ -62,6 +62,15 @@ def self.time_exceeded
     TaskStatus.find(12)
   end
 
+  # Return the count (which equals the largest id) - so that other code can loop thought all statuses without database lookup
+  #
+  # Make sure to update this if/when you add another status!
+  #
+  # Keep this hard coded! Saves cache load time.
+  # Important: count must equal the largest id in the database
+  def self.count
+    12
```

Comment:
**thought:** I think this can be improved. However, it seems that we are hardcoding other index values already (e.g. `TaskStatus.find(12)`), this might be a good compromise for now.

Reply:
Yes - I was thinking the same. So we will leave this as is... it is probably good to add a TODO comment in there as something to consider later.